Components/ImportService.php

```diff
@@ -544,4 +543,80 @@ private function getAst($categoryId, $query = '', $showOnlyConnectArticles = tru
 
         return $ast;
     }
+
+    /**
+     * @param array $articleIds
+     * @param $categoryId
+     */
+    private function unAssignArticlesFromCategory(array $articleIds, $categoryId)
+    {
+        $categories = [];
+        $categories[] = $categoryId;
+        $childCategories = $this->manager->getConnection()->executeQuery('SELECT id FROM s_categories WHERE path LIKE ?',
+            ["%|$categoryId|%"]);
+
+        while ($childCategory = $childCategories->fetchColumn()) {
+            $categories[] = (int) $childCategory;
+        }
+
+        $categoriesStatement = $this->manager->getConnection()->prepare('DELETE FROM s_articles_categories WHERE articleID IN (' . implode(', ', $articleIds) . ') AND categoryID IN (' . implode(', ', $categories) . ')');
```

Comment:
Don't know why, but doesn't work when try to run it manually.
I saw that tests are reaching this line.
have you ran it in your shopware?
I'm trying to run ```EXPLAIN DELETE FROM s_articles_categories WHERE articleID IN (5,1,2,3) AND categoryID IN (5,1,2,3);``` and got following error: 10:25:51 Kernel error: Error( 1064 ) 42000: "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DELETE FROM s_articles_categories WHERE articleID IN (5,1,2,3) AND categoryID IN' at line 1".

if you want to use where IN, you need to do nested queries I think.
Btw 
```
if ($categoryId !== null) {
 +                    $this->unAssignArticlesFromCategory($articleIds, $categoryId);
 +                } else {
 +                    $this->unAssignArticlesFromAllCategories($articleIds);
 +                }
``` 
is wrapped in transaction. You know articleIds and categoryIds including child categories. Then you can call delete query for combination of every article & category. At the end they'll be commited at once.

Reply:
I can run it locally 
![image](https://user-images.githubusercontent.com/15930605/31007247-38599a0e-a500-11e7-8dff-757d17626df2.png).
I don't understand what you mean in the second part